lib/rbac/filterer.rb

```diff
@@ -514,15 +514,18 @@ def scope_for_user_role_group(klass, scope, miq_group, user, managed_filters)
       if user_or_group.try!(:self_service?) && MiqUserRole != klass
         scope.where(:id => klass == User ? user.id : miq_group.id)
       else
+        role = user_or_group.miq_user_role
         # hide creating admin group / roles from non-super administrators
-        unless user_or_group.miq_user_role&.super_admin_user?
+        unless role&.super_admin_user?
           scope = scope.with_roles_excluding(MiqProductFeature::SUPER_ADMIN_FEATURE)
         end
 
         if MiqUserRole != klass
           filtered_ids = pluck_ids(get_managed_filter_object_ids(scope, managed_filters))
           # Non admins can only see their own groups
-          scope = scope.with_groups(user.miq_group_ids) unless user_or_group.miq_user_role&.super_admin_user?
+          if !role&.super_admin_user? && !role&.tenant_admin_user?
```

Comment:
Since a super admin is also a tenant admin, I think you want:

```ruby
scope = scope.with_groups(user.miq_group_ids) unless role&.tenant_admin_user?
```

The test I suggested ensures that `super_user.tenant_admin_user? == true`

Reply:
Interesting... great point, it's true.  Do we lose clarity by removing this though?  I'm good with using this suggestion.